test/test_front_matter_defaults.rb

```diff
@@ -53,9 +49,36 @@ class TestFrontMatterDefaults < JekyllUnitTest
       assert_equal "val", @affected.data["key"]
       assert_nil @not_affected.data["key"]
     end
+  end
+
+  context "A site with collections and front matter defaults with glob patterns" do
+    setup do
+      site = fixture_site(
+        "collections_dir" => "gathering",
+        "collections"     => { "staff" => { "output" => true } },
+        "defaults"        => [
+          {
+            "scope"  => { "path" => "_staff/**/*.md", "type" => "staff" },
+            "values" => { "layout" => "simple" },
+          },
+          {
+            "scope"  => { "path" => "_staff/**/*.svg" },
+            "values" => { "css_class" => "epilson" },
+          },
+        ]
+      )
+      site.read
+      @staff = site.collections["staff"]
+    end
+
+    should "affect the appropriate items only" do
+      item = @staff.docs.sample
```

Comment:
[`Array#sample`](https://ruby-doc.org/core-3.1.1/Array.html#method-i-sample) selects a random item, so if you're expecting to always have the same item come out, then use `#shift` or `#[]`.

Reply:
Actually, the intention here is to indeed test a random object in the *homogeneous* array during each run.

Reply:
Every doc is the same, or the array is of size 1? I’m still confused about this. If it’s one element, it would be clearer to first assert the size, then assert its contents. If the array has >1 element that you expect to be the same, wouldn’t it be more effective to iterate over all elements and assert each element has the contents you want.

what I’m afraid of is that we’ll add a new doc or file to this collection in the fixture site and these tests will randomly fail rather than consistently fail. I think tests should be deterministic to avoid contributors chasing down confusing errors that only occur sometimes.